File: Python/0371_sum_of_two_integers.py
# 2020-08-04 00:38:23

class Solution:
    def getSum(self, a: int, b: int) -> int:
        return (a if a < 0x7ffffff else a - 0x100000000) \
            if b == 0 \
            else self.getSum((a ^ b) & 0xffffffff,
                             ((a & b) << 1) & 0xffffffff)

File: Java/0371_Sum_of_Two_Integers.java
// 2020-08-04 00:00:59

class Solution {
    public int getSum(int a, int b) {
        int sum = a, carry = b;
        while (carry != 0) {
            sum = a ^ b;
            carry = (a & b) << 1;
            a = sum;
            b = carry;
        }
        return sum;
    }
}